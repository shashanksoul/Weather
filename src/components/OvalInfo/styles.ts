import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  container: {
    padding: 20,
    backgroundColor: 'white',
    borderRadius: 40,
    width: '100%',
    margin: 15,
    paddingLeft: 30,
  },
  dangerBg: {
    backgroundColor: 'red',
  },
  heading: {
    fontFamily: 'SpaceMono-Bold',
    fontSize: 16,
    color: 'black',
  },
  subHeading: {
    fontSize: 14,
    color: 'black',
  },
});

export default style;
