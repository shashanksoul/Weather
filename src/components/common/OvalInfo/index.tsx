import React from 'react';
import {Text, View} from 'react-native';

import style from './styles';

type OvalInfoProps = {
  heading: string;
  subHeading: string;
  isDanger?: boolean;
};
const OvalInfo: React.FC<OvalInfoProps> = ({
  heading,
  subHeading,
  isDanger = false,
}) => {
  return (
    <View
      testID="oval-view"
      style={isDanger ? [style.container, style.dangerBg] : style.container}>
      <Text
        testID="heading-text"
        style={isDanger ? [style.heading, {color: 'white'}] : style.heading}>
        {heading.toUpperCase()}
      </Text>
      <Text
        testID="subHeading-text"
        style={
          isDanger ? [style.subHeading, {color: 'white'}] : style.subHeading
        }>
        {subHeading}
      </Text>
    </View>
  );
};

export default OvalInfo;
