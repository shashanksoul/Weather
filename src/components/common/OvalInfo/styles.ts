import {StyleSheet} from 'react-native';

import {SCREEN_WIDTH} from 'src/utils/Dimensions';

const style = StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: 'white',
    borderRadius: SCREEN_WIDTH / 2,
    width: '100%',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    margin: 10,
    paddingHorizontal: 20,
  },
  dangerBg: {
    backgroundColor: 'red',
  },
  heading: {
    fontFamily: 'SpaceMono-Bold',
    fontSize: 14,
    color: 'black',
  },
  subHeading: {
    fontSize: 13,
    color: 'black',
  },
});

export default style;
