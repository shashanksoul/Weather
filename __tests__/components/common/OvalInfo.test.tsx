import React from 'react';

import OvalInfo from '@components/common/OvalInfo';
import {render} from '@testing-library/react-native';

const mockHeading = 'This is Heading';
const mockSubHeading = 'This is subHeading';
describe('OvalInfo Component', () => {
  test('render', () => {
    const ovalInfo = render(
      <OvalInfo heading={mockHeading} subHeading={mockSubHeading} />,
    );
    const heading = ovalInfo.getByTestId('heading-text');
    const subHeading = ovalInfo.getByTestId('subHeading-text');
    expect(heading).toBeTruthy();
    expect(subHeading).toBeTruthy();
    expect(ovalInfo.toJSON()).toMatchSnapshot();
  });
  test('when isDanger is true', () => {
    const ovalInfo = render(
      <OvalInfo isDanger heading={mockHeading} subHeading={mockSubHeading} />,
    );
    const oval = ovalInfo.getByTestId('oval-view');
    const heading = ovalInfo.getByTestId('heading-text');
    expect(oval.props.style[1].backgroundColor).toEqual('red');
    expect(heading.props.style[1].color).toEqual('white');
  });
});
