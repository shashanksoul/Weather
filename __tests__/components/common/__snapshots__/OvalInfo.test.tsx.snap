// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OvalInfo Component render 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "borderRadius": 375,
      "justifyContent": "flex-start",
      "margin": 10,
      "padding": 16,
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
  testID="oval-view"
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "SpaceMono-Bold",
        "fontSize": 14,
      }
    }
    testID="heading-text"
  >
    THIS IS HEADING
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 13,
      }
    }
    testID="subHeading-text"
  >
    This is subHeading
  </Text>
</View>
`;
